### PR TITLE
frontend popup

### PR DESCRIPTION
**Pull request by Arthurllyr**

### CODE REVIEW

**Comment by cedricserugendo** [comment]
`showInfos` seems to be a function that only returns a Boolean and not an object 
Did you check the returned result?
@Arthurllyr

**Comment by Arthurllyr** [comment]
Indeed, showInfos seems to unlock the popup when there is some datas to show.

File: frontend/src/src/components/CMList.svelte [hunk 1]
```diff
@@ -5,8 +5,8 @@
   import {areaSelectionLayerStore, selectedLayerStore, isCMPaneActiveStore} from '../stores.js';
   import CM from './CM.svelte';
   import AreaSelection from './AreaSelection.svelte';
-// import PopupContent from '../leaflet_components/L.TileLayer.QueryableLayer.js';
-// import '../leaflet_components/L.TileLayer.QueryableLayer.css';
+//  import PopupContent from '../leaflet_components/L.TileLayer.QueryableLayer.js';
+//  import '../leaflet_components/L.TileLayer.QueryableLayer.css';
```
Comment (cedricserugendo) [inline frontend/src/src/components/CMList.svelte:9]: you should uncomment this lines

File: frontend/src/src/components/CMList.svelte [hunk 2]
```diff
@@ -2,11 +2,10 @@
   import {onMount} from 'svelte';
   import {BASE_URL} from '../settings.js';
   import {getCMs} from '../client.js';
-  import {areaSelectionLayerStore, selectedLayerStore, isCMPaneActiveStore} from '../stores.js';
+  import {areaSelectionLayerStore, selectedLayerStore, isCMPaneActiveStore, popupInformation} from '../stores.js';
   import CM from './CM.svelte';
   import AreaSelection from './AreaSelection.svelte';
-// import PopupContent from '../leaflet_components/L.TileLayer.QueryableLayer.js';
-// import '../leaflet_components/L.TileLayer.QueryableLayer.css';
+  // import {popupContent} from '../leaflet_components/L.TileLayer.QueryableLayer.js';
```
Comment (cedricserugendo) [inline frontend/src/src/components/CMList.svelte:8]: this line must be removed

File: frontend/src/src/components/CMList.svelte [hunk 3]
```diff
@@ -158,9 +159,11 @@
...
         {/each}
       {:else if activeTabTest === 'consultation'}
-        <p> Information about the house. </p>
-        // plot popupContent here
-
+        <div class="popupInformation">
+            <table>
+                {@html $popupInformation}
+            </table>
+        </div>
```
Comment (cedricserugendo) [inline frontend/src/src/components/CMList.svelte:166]: by default you show all the information in the HTLM string, we want to show only few information, this part need modification

File: frontend/src/src/components/Map.svelte [hunk 4]
```diff
@@ -136,7 +137,7 @@
         } else {
           title = layer.labels.primary;
         }
-
+        // appel des datas contenues dans l'objet showInfos
```
Comment (cedricserugendo) [inline frontend/src/src/components/Map.svelte:140]: this commented line is not useful I think

File: frontend/src/src/leaflet_components/L.TileLayer.QueryableLayer.js [hunk 5]
```diff
@@ -1,3 +1,9 @@
+// import {writable} from 'svelte/store';
+// export const popupContent  = writable(true);
```
Comment (cedricserugendo) [inline frontend/src/src/leaflet_components/L.TileLayer.QueryableLayer.js:2]: not usefull

File: frontend/src/src/leaflet_components/L.TileLayer.QueryableLayer.js [hunk 6]
```diff
@@ -26,16 +32,20 @@ const BaseMethods = {
         });
   },
 
+  //  création de l'objet showInfos
+
```
Comment (cedricserugendo) [inline frontend/src/src/leaflet_components/L.TileLayer.QueryableLayer.js:36]: not usefull

File: frontend/src/src/leaflet_components/L.TileLayer.QueryableLayer.js [hunk 7]
```diff
@@ -26,16 +32,20 @@ const BaseMethods = {
...
   showInfos: function(title, latlng, content) {
     // Otherwise show the content in a popup, or something.
     if (!content || !content.features || (content.features.length == 0)) {
       return false;
     }
 
+    //   declare variable popupContent à  l'interieur de l'objet showInfos
+
```
Comment (cedricserugendo) [inline frontend/src/src/leaflet_components/L.TileLayer.QueryableLayer.js:44]: not usefull

File: frontend/src/src/leaflet_components/L.TileLayer.QueryableLayer.js [hunk 8]
```diff
@@ -26,16 +32,20 @@ const BaseMethods = {
...
     let popupContent = '';
     const allFields = {};
 
-    popupContent += '<h1>' + title + '</h1>';
+    popupContent += '<h3>' + title + '</h3>';
```
Comment (cedricserugendo) [inline frontend/src/src/leaflet_components/L.TileLayer.QueryableLayer.js:48]: why have you change the tag ? according to me it's not necessarry

**Review by cedricserugendo — CHANGES_REQUESTED** [review]

**Comment by cedricserugendo** [comment]
@Arthurllyr, you can comment the PR like this as discussed

**Comment by Arthurllyr** [comment]
> @Arthurllyr, you can comment the PR like this as discussed

do you understand the difference btw titles and rows ?

**Comment by cedricserugendo** [comment]
> > @Arthurllyr, you can comment the PR like this as discussed
> 
> do you understand the difference btw titles and rows ?

The difference is clear, but can you send me a screenshot of the error you have with the latest version of the code?

**Comment by Arthurllyr** [comment]
> > > @Arthurllyr, you can comment the PR like this as discussed
> > 
> > 
> > do you understand the difference btw titles and rows ?
> 
> The difference is clear, but can you send me a screenshot of the error you have with the latest version of the code?
Capture d’écran 2022-07-05 195905

**Comment by cedricserugendo** [comment]
> > > @Arthurllyr, you can comment the PR like this as discussed
> > 
> > 
> > do you understand the difference btw titles and rows ?
> 
> The difference is clear, but can you send me a screenshot of the error you have with the latest version of the code? Capture d’écran 2022-07-05 195905

we have already encountered this error 
You try to apply a 'length' method on an undefined object

**Comment by Arthurllyr** [comment]
popupcontent

**Review by cedricserugendo — APPROVED** [review]
from my side, this branch can be merged if Gilles and you, @Arthurllyr , are ok with that

**Pull request merged**